src/ad8232/ad8232.cxx

```diff
@@ -34,11 +34,31 @@ AD8232::AD8232(int loPlus, int loMinus, int output, float aref) :
 {
   m_gpioLOPlus.dir(mraa::DIR_IN);
   m_gpioLOMinus.dir(mraa::DIR_IN);
-  
+
   m_aref = aref;
   m_ares = (1 << m_aioOUT.getBit());
 }
 
+AD8232::AD8232(std::string initStr) : mraaIo(initStr), m_gpioLOPlus(mraaIo.gpios[0]),
```

Comment:
Initialize m_* members inside the function because you can accidentally access elements outside the array(garbage) with the direct indexing. Do some sanity checks for them inside the function's body.